Sources/Tracking/Background Queue/QueueRunRequest.swift

```diff
@@ -53,7 +65,8 @@ public class CioQueueRunRequest: QueueRunRequest {
             let success = storage.delete(storageId: nextTaskStorageId)
             logger.debug("deleted task \(nextTaskStorageId) success: \(success)")
 
-            return goToNextTask(query: query, queryTotalNumberTasks: queryTotalNumberTasks)
+            return goToNextTask(query: query, queryTotalNumberTasks: queryTotalNumberTasks,
+                                lastFailedTask: nextTaskToRunInventoryItem)
```

Comment:
seems like this case should get a `nil` for `lastFailedTask` because it is trashed and shouldn't be considered as blocking criteria

Reply:
I didn't use `nil` because if a task can't be found in persistent storage, then it's a failed execution just like failing a HTTP request. 

I do believe that we should mark this task not found as `lastFailedTask`, but we should *not* be deleting tasks from persistent storage if it's not found. 

The way the queue is designed today is that *individual tasks* are only deleted if they run successfully. In the future we may decide to also delete stale tasks in the queue but we should be deleting *groups* of stale tasks.

Reply:
Made change in https://github.com/customerio/customerio-ios/pull/93/commits/f314e936c330e83ef36b6041c3492e76152ab91d

Reply:
I think this results in a stuck point in the inventory, no? if Y and Z depend on task X that can't be found, we should stop trying to find X and just let Y & Z continue, unless there's some way in which we expect X may be found on a subsequent run?

Reply:
Let's say that X is the start of a group and Y and Z are members of the group. So, Y and Z depend on X running successfully. 

If X cannot be found in the file system, then we should *not* attempt to run Y or Z since they depend on X. 

Let's say that X ran successfully but Y was not found in the file system. Z will still execute because the queue run request code only skips a task if *the a task that failed was the start of a group*. 

So, the queue will only be *stuck* here if X is not found in the file system and X is the start of a group. In the case, we should come back again to the idea of deleting stale tasks in the queue.